Source/DafnyLanguageServer/Workspace/TextDocumentLoader.cs

```diff
@@ -1,51 +1,90 @@
-﻿using System;
-using IntervalTree;
+﻿using IntervalTree;
 using Microsoft.Dafny.LanguageServer.Language;
 using Microsoft.Dafny.LanguageServer.Language.Symbols;
 using Microsoft.Dafny.LanguageServer.Workspace.Notifications;
 using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using System.Collections.Concurrent;
 using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace Microsoft.Dafny.LanguageServer.Workspace {
+  /// <summary>
+  /// Text document loader implementation that offloads the whole procedure on one dedicated
+  /// thread with a stack size of 256MB.
+  /// </summary>
+  /// <remarks>
+  /// The increased stack size is necessary to solve the issue https://github.com/dafny-lang/dafny/issues/1447.
+  /// </remarks>
   public class TextDocumentLoader : ITextDocumentLoader {
-    private readonly IDafnyParser _parser;
-    private readonly ISymbolResolver _symbolResolver;
-    private readonly IProgramVerifier _verifier;
-    private readonly ISymbolTableFactory _symbolTableFactory;
-    private readonly ICompilationStatusNotificationPublisher _notificationPublisher;
+    // 256MB
+    private const int MaxStackSize = 0x10000000;
 
-    public TextDocumentLoader(
+    private readonly IDafnyParser parser;
+    private readonly ISymbolResolver symbolResolver;
+    private readonly IProgramVerifier verifier;
+    private readonly ISymbolTableFactory symbolTableFactory;
+    private readonly ICompilationStatusNotificationPublisher notificationPublisher;
+    private readonly BlockingCollection<LoadRequest> loadRequests = new();
```

Comment:
Does `BlockingCollection` provide any guarantees about the order in which elements are taken out? Consider using `ConcurrentQueue` to make this more clear.